ENH: Add CITATION.cff and prepare Zenodo integration

Prachi194agrawal opened the pull request:
Reference Issue: Fixes #1334

New Tests Added: NA

Documentation Updated: Yes

Change Log Entry: Add CITATION.cff and Zenodo release documentation

Details
Description
This PR prepares the repository for Zenodo integration as discussed in issue #1334.

Changes
Added CITATION.cff: Configured to use the JMLR 2021 paper (Feurer et al.) as the preferred-citation. This ensures users citing the software are directed to the correct academic publication, rather than just the source code.

Updated README: Added a placeholder for the Zenodo DOI badge in the header.

Updated Documentation: Added a note in CONTRIBUTING.md (Appendix) regarding the Zenodo release workflow to guide future maintainers.

Prachi194agrawal commented:
@PGijsbers This PR is ready for review!

It includes the `CITATION.cff` file, the documentation updates, and the placeholder badge in the README as discussed.

Once this is merged, a repository admin will just need to enable the repository in the Zenodo settings to start archiving future releases.

PGijsbers commented on CITATION.cff:
```diff
@@ -1,5 +1,17 @@
+  - family-names: "Gijsbers"
+    given-names: "Pieter"
+  - family-names: "Vanschoren"
+    given-names: "Joaquin"
```
Why are (only) these three people chosen? What is supposed to be in this field if we already reference the preferred-citation?

Prachi194agrawal commented:
@PGijsbers Thanks for the review! I have updated the PR based on your feedback:

1. **Scope:** I removed the unrelated `edit_flow` files so this PR focuses only on the Citation work.
2. **Abstract:** I restored the full abstract text in `CITATION.cff`.
3. **Authors:** As mentioned in my inline reply, I am waiting for your guidance on whether to list all software contributors or use a collective name like "The OpenML Team".

Ready for re-review!
